htlcswitch/hop/iterator.go

```diff
@@ -109,7 +109,7 @@ func (r *sphinxHopIterator) HopPayload() (*Payload, error) {
 		isFinal := r.processedPacket.Action == sphinx.ExitNode
 		payload, parsed, err := NewPayloadFromReader(
 			bytes.NewReader(r.processedPacket.Payload.Payload),
-			isFinal,
+			isFinal, r.blindingKit.UpdateAddBlinding.IsSome(),
```

Comment:
> ValidateParsedPayloadTypes is probably a good spot for this check.

Would need to propagate information about whether we had a blinding point in `update_add_htlc` all the way down to do this here (eg, a `isRelayingBlinded bool`). Is that preferable over calling it in `HopPayload`, since we've already decided to do blinded path "post-processing" there with `DecryptAndValidateFwdInfo` - no strong opinions!